### ISSUE TITLE
Show progress whilst processing image

### ISSUE THREAD

**Issue by RawConvert** [darktable-org/darktable#6373]
The suggestion is that whilst waiting for the image to be processed, lighttable could show progress by displaying the name of the module executing.  E.g. next to the coloured labelling dots near the bottom of the window.  The names would be the actual module name, not some internal routine id.
It would be interesting to see how long the various modules execute for; possibly useful for understanding GPU usage?

**Comment by johnny-bit** [comment]
Why would this be useful? There's little to be gained. And actual, useful performance stats are  available when starting drama with `-d perf`flag.
IMO this would just be noise.

**Comment by dterrahe** [comment]
I have actually been thinking for a long time (basically since I started using dt) that the "working..." text in the middle of the picture is not a very good feedback ui. It doesn't stand out enough (depending on the image of course), isn't large enough and requires the eye to move from where one is currently working (either in the module panel or, when using accelerators, on another area of the image. Personally, I would prefer a visual indication by, say, a change in border color. And if this could progress clockwise around to image to indicate percentage complete, even better. I did some experimenting a while back; creating the border was easy enough, but the pixel pipe renderer doesn't seem to get called enough to produce a smooth progress indicator.

I don't necessarily need a breakdown by module, as suggested here, but some form of indication of #modules left would be useful/entertaining.

**Comment by johnny-bit** [comment]
@dterrahe - well, "working" in darkroom is unobtrusive communication that something is delaying displaying of processed image (like intensive task). I remember experimenting with progressbar on something else but that's imo visual mess.

**Comment by dterrahe** [comment]
_unobtrusive_ is key word here; you won't see it unless you look straight at it. Which means if you are waiting for a long operation to finish, you are going to be staring at it all the time. Otherwise the only way to be sure it is done (there might unexpectedly be little impact from a subtle slider move) is if you search and can't find the "working" anymore. No "corner of eye" thing here; it is much too unobtrusive for that!

**Comment by junkyardsparkle** [comment]
> No "corner of eye" thing here; it is much too unobtrusive for that!

With the transparent background, it kind of depends on what kind of imagery happens to be behind it. I wonder if just giving it an opaque background box would be enough to improve the situation?

**Comment by difrkaguilar** [comment]
Maybe something better than the unobtrusive word **"working"** would be use a tiny and unobtrusive too progress bar at the top left of the image "if" the task is longer than "x" milliseconds
Like the progress bar at the lower left side to export images in the lightview.
In that case, users can know the progress of the task.
See attached image.
![progresBar01](https://user-images.githubusercontent.com/67672868/94222017-4a2c2280-feba-11ea-9752-d80d5768e2c4.jpg)

**Comment by TurboGit** [comment]
Yes for CPU usage we have -d perfs and this wouldn't be very useful in GUI as most of the time on a standard computer the pipe is executing very fast. You'll certainly not be able to read 10 labels or more in less than a second!

**Comment by RawConvert** [comment]
@johnny-bit ,  Yes noise for some I suppose.  I see it rather like the rev counter in a car, not essential, lots of drivers don't ever look at it I'm sure, but it's quite nice to have available.

@dterrahe ,   Is it possible to show % complete? - I imagine it's quite hard for the s/ware to know how long in total it's going to take?

@TurboGit ,   I have a decent PC (i7 processor) but processing seldom takes a second.  With a couple of masks and using the highlight recovery iterations in filmic, it can take tens of seconds.  Presumably some modules would flash by and you couldn't read the names - but you're likely not too interested in these anyway.  You'd be able to see the longer running ones.

**Comment by AndreaMonzini** [comment]
> I have actually been thinking for a long time (basically since I started using dt) that the "working..." text in the middle of the picture is not a very good feedback ui... 

Actually, in my experiencee, "Working..." is very distracting.
In particular with a fast machine i keep seeing "working" continuously in the middle of the image  even with a near realtime processing but i understand it can be useful to know when long processing is not finished.
Maybe we could add the option to hide "working..."  and/or we could show  "working..." only with longer processing.
